clash-cores/src/Clash/Cores/Xilinx/Ethernet/Gmii.hs

```diff
@@ -0,0 +1,117 @@
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE ViewPatterns #-}
+
+{- |
+Copyright   :  (C) 2024, Google LLC
+License     :  BSD2 (see the file LICENSE)
+Maintainer  :  QBayLogic B.V. <devops@qbaylogic.com>
+
+Contains a wrapper for the Xilinx GMII to SGMII PMA core :
+ "LogiCORE IP Ethernet 1000BASE-X PCS/PMA or SGMII"
+ https://www.xilinx.com/xsw/gig_ethernet_pcs_pma
+
+Currently only the Verilog primitive is supported.
+-}
+module Clash.Cores.Xilinx.Ethernet.Gmii (
+  --- Types
+  AutoNegConfig (..),
+  BridgeOutput (..),
+  Config (..),
+  DuplexMode (..),
+  Gmii (..),
+  LinkSpeed (..),
+  Lvds (..),
+  Nanoseconds,
+  Pause (..),
+  Picoseconds,
+  Status (..),
+  --- Gmii to Sgmii bridge
+  gmiiSgmiiBridge,
```

Comment:
I don't understand the structure of the Haddock; it seems primarily based on alphabetic order? I think this is quite an improvement:
```suggestion
  -- * GMII SGMII bridge function
  gmiiSgmiiBridge,
  -- * GMII interface
  Gmii (..),
  -- * GMII SGMII bridge function output
  BridgeOutput (..),
  -- ** Status fields
  Status (..),
  Pause (..),
  -- * PHY LVDS signals
  Lvds (..),
  -- * Bridge configuration
  Config (..),
  AutoNegConfig (..),
  DuplexMode (..),
  LinkSpeed (..),
```
Note that I no longer export `Nanoseconds` and `Picoseconds` because I dont' see why you'd do that and I didn't want to complicate the job of organising the Haddock even more by including stuff that's not going to be included anyway.

I also removed the two `---` comments as I don't know what they are, as I note in a different review comment.